src/components/DealForm.tsx

```
@@ -24,169 +23,51 @@ interface DealFormProps {
 }
 
 const DealForm: FunctionComponent<DealFormProps> = ({ onSubmit }) => {
+	const [currentStep, setCurrentStep] = useState<number>(0);
 	const [form] = Form.useForm();
 	const intl = useIntl();
+	const steps = [
+		intl.formatMessage({
+			defaultMessage: "Deal details",
+			description: "Deal form: deal details step title",
+		}),
+		intl.formatMessage({
+			defaultMessage: "Tranche structure",
+			description: "Deal form: tranche structure step title",
+		}),
+		intl.formatMessage({
+			defaultMessage: "Review",
+			description: "Deal form: review step title",
+		}),
+	];
 
-	const validateMinPrincipal = (value): Promise<void> => {
-		const validationMessage = intl.formatMessage({
-			defaultMessage: "'principal' needs to be greater than 0",
-			description: "Deal form: min principal validation message",
-		});
-		return validateMinValue(value, 0, validationMessage);
-	};
+	const showStep = (step: number) => {
+		if (step === currentStep) {
+			return "block";
+		}
 
-	const validateMinTimeToMaturity = (value): Promise<void> => {
-		const validationMessage = intl.formatMessage({
-			defaultMessage: "'time to maturity' needs to be greater than 0",
-			description: "Deal form: min time to maturity validation message",
-		});
-		return validateMinValue(value, 0, validationMessage);
+		return "hidden";
 	};
 
-	const validateBorrowerKey = (value): Promise<void> => {
-		const validationMessage = intl.formatMessage({
-			defaultMessage: "'borrower key' isn't valid",
-			description: "Deal form: borrower key validation message",
-		});
-		return validatePublicKey(value, validationMessage);
+	const onNextStep = async (fieldsToValidate: string[], nextStep: number) => {
+		try {
+			await form.validateFields(fieldsToValidate);
```

`await form.validateFields(fieldsToValidate).then(() => setCurrentStep(nextStep))`

---

no try empty catch needed